doc/BGP/BGP-route-aggregation-with-bbr-awareness.md

```diff
@@ -0,0 +1,310 @@
+4. Disable BBR feature in config DB:
+    - In config DB, find out all addresses that has bbr-required equals true and remove aggregated addresses in the bgp container then update state DB with inactive state.
+5. The bgp container restarted:
+    - First, the bgp container will clean the addresses in state DB and then the bgp container will process all existed config one by one according to 1~4.
+
+To avoid concurrency issue, all operation mentioned above will be put into queue and be processed one by one. 
+#### UML Sequence Diagrams
+![](img/add-aggregated-address.png)
+
+*Fig.1 Add Address*
+
+![](img/remove-aggregated-address.png)
+
+*Fig.2 Remove Address*
+
+![](img/bbr-feature-state-change.png)
+
+*Fig.3 BBR Feature State Change*
+
+### CLI Design
+  | CLI |               Description                        |
+  |:----------------------|:-----------------------------------------------------------|
+  | show ip bgp aggregate-address | Show aggregate address in ipv4 address family |
+  | show ipv6 bgp aggregate-address | Show aggregate address in ipv6 address family |
+  | config bgp aggregate-address add | Add aggregate address |
+  | config bgp aggregate-address remove | Remove aggregate address |
+
+#### Show CLI
+**show ip bgp aggregate-address**
+
+This command is used to show aggregate address in ipv4 address family
+- Usage
+  ```
+  show ip bgp aggregate-address
+  ```
+
+- Example
+  ```
+  > show ip bgp aggregate-address
+  +----------------+---------+-------------+-------------+---------------+
+  |Prefix          |State    |Summary Only |BBR Required |As Set Enabled |
+  |----------------+---------+-------------+-------------+---------------+
+  |192.168.0.0/24  |Active   |False        |False        |True           |
+  +----------------+---------+-------------+-------------+---------------+
+  |10.0.0.0/24     |Inactive |True         |True         |False          |
+  +----------------+---------+-------------+-------------+---------------+
+  ```
+
+**show ipv6 bgp aggregate-address**
+
+This command is used to show aggregate address in ipv6 address family
+- Usage
+  ```
+  show ipv6 bgp aggregate-address
+  ```
+
+- Example
+  ```
+  > show ipv6 bgp aggregate-address
+  +----------------+---------+-------------+-------------+---------------+
+  |Prefix          |State    |Summary Only |BBR Required |As Set Enabled |
+  |----------------+---------+-------------+-------------+---------------+
+  |fc00:1::/64     |Active   |False        |False        |True           |
+  +----------------+---------+-------------+-------------+---------------+
+  |fc00:3::/64     |Inactive |True         |True         |False          |
+  +----------------+---------+-------------+-------------+---------------+
+  ```
+
+#### Config CLI
+**config bgp aggregate-address add**
+
+This command is used to add aggregate address
+- Usage
+  ```
+  config bgp aggregate-address add <address> [--bbr-required] [--summary-only] [--as-set]
```

Comment:
Does it have default value for these optional param? Means explicitly enabled with param and disabled w/o param?

Reply:
Default value is false,
those parameters are flag, if added it will be true, else be false.